app/jobs/regular/change_display_name.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChangeDisplayName < ::Jobs::Base
+    sidekiq_options queue: "low"
+
+    def execute(args)
+      @user = User.find_by(id: args[:user_id])
+
+      return unless user.present?
+
+      # We need to account for the case where the instance allows
+      # name to be empty by falling back to username.
+      @old_display_name = (args[:old_name].presence || user.username).unicode_normalize
+      @new_display_name = (args[:new_name].presence || user.username).unicode_normalize
+
+      @quote_rewriter = QuoteRewriter.new(user.id)
+
+      update_posts
+      update_revisions
+    end
+
+    private
+
+    attr_reader :user, :old_display_name, :new_display_name, :quote_rewriter
+
+    def update_posts
+      Post
+        .with_deleted
+        .joins(quoted("posts.id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |post| update_post(post) }
+    end
+
+    def update_revisions
+      PostRevision
+        .joins(quoted("post_revisions.post_id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |revision| update_revision(revision) }
+
+      PostRevision
+        .joins(quoted("post_revisions.post_id"))
+        .where("p.user_id = :user_id", user_id: user.id)
+        .find_each { |revision| update_revision(revision) }
+    end
+
+    def update_post(post)
+      post.raw = update_raw(post.raw)
+      post.cooked = update_cooked(post.cooked)
+
+      post.update_columns(raw: post.raw, cooked: post.cooked)
+
+      SearchIndexer.index(post, force: true) if post.topic
+    rescue => e
+      Discourse.warn_exception(e, message: "Failed to update post with id #{post.id}")
+    end
+
+    def update_revision(revision)
+      if revision.modifications.key?("raw") || revision.modifications.key?("cooked")
+        revision.modifications["raw"]&.map! { |raw| update_raw(raw) }
+        revision.modifications["cooked"]&.map! { |cooked| update_cooked(cooked) }
+        revision.save!
+      end
```

Comment:
```suggestion
      if revision.modifications["raw"] || revision.modifications["cooked"]
        revision.modifications["raw"].map! { |raw| update_raw(raw) }
        revision.modifications["cooked"].map! { |cooked| update_cooked(cooked) }
        revision.save!
      end
```

I think we can avoid the lonely operator here if we just check for the value in the conditional

Reply:
I was hesitant at first, but I guess this is technically safe, since there should be no case where there are modifications to only one of `raw` and `cooked`. 👍 Do you think we should update the conditional to use `&&` as a soft of assertion here?

Reply:
O sorry I didn't notice that the conditional is `||` and not `&&`. I think we can leave this it is.